File: Algorithms/sorting/bubbleSort.py
def bubbleSort(A, n):
   for i in range(n):
      flag = 0
      for j in range(n-i-1):
         if A[j] > A[j+1]:
            A[j], A[j+1] = A[j+1], A[j]
            flag = 1

      if(flag == 0):
         break

if __name__ == "__main__":
    A = [0, 8, 90, 890, 4, 45, 7]
    n = 7
    bubbleSort(A, n)
    print(A)

File: Algorithms/sorting/bubbleSort.cpp
#include <bits/stdc++.h>
using namespace std;

void bubbleSort(int A[], int n)
{
   for (int i = 0; i < n; i++)
   {
      int flag = 0;
      for (int j = 0; j < n - i - 1; j++)
      {
         if (A[j] > A[j + 1])
         {
            int temp = A[j];
            A[j] = A[j + 1];
            A[j + 1] = temp;
            flag = 1;
         }
      }
      if (flag == 0)
         break;
   }
}
int main()
{
   int A[] = {4,
              5,
              78,
              9,
              420,
              33};
   int n = 6;
   bubbleSort(A, n);
   for (int i = 0; i < n; i++)
   {
      cout << A[i] << " ";
   }

   return 0;
}